cpp/src/gandiva/precompiled/time.cc

```diff
@@ -689,21 +689,46 @@ timestamp castTIMESTAMP_utf8(int64_t context, const char* input, int32 length) {
 
 timestamp castTIMESTAMP_date64(date64 date_in_millis) { return date_in_millis; }
 
-char* castVARCHAR_timestamp_int64(int64 context, timestamp in, int64 length,
-                                  int32* out_len) {
-  std::string timestamp_str = std::to_string(in);
+const char* castVARCHAR_timestamp_int64(int64 context, timestamp in, int64 length,
+                                        int32* out_len) {
+  long long int year = static_cast<long long int>(extractYear_timestamp(in));
+  long long int month = static_cast<long long int>(extractMonth_timestamp(in));
+  long long int day = static_cast<long long int>(extractDay_timestamp(in));
+  long long int hour = static_cast<long long int>(extractHour_timestamp(in));
+  long long int minute = static_cast<long long int>(extractMinute_timestamp(in));
+  long long int second = static_cast<long long int>(extractSecond_timestamp(in));
+  long long int millis = static_cast<long long int>(in % MILLIS_IN_SEC);
+
+  const int full_output_length = 23;
+  const int char_buffer_length = full_output_length + 1;  // snprintf adds \0
+  char char_buffer[char_buffer_length];
+
+  // yyyy-MM-dd hh:mm:ss.sss
+  snprintf(char_buffer, char_buffer_length,
```

Comment:
Do note that `snprintf` is locale-dependent. I'm not sure this makes a difference for pure integer formatting, but it's something to be careful about in general (especially if formatting floating-point numbers).

Reply:
one alternative would be std::setw/std::setfill. but, we've had trouble with streams in the precompiled library earlier. however, that may be gone now (we don't statically link libstdc++ any longer).

@pprudhvi, can you please give this a try ?

Reply:
C++ streams will have the same issue by default (though you can call [imbue](https://en.cppreference.com/w/cpp/io/basic_ios/imbue) to change the stream's locale).

Reply:
since we are using this only for integers, this should be fine.